app/controllers/api/v2/user_controller.rb

```diff
@@ -7,6 +7,37 @@ def login_required
     head :unauthorized unless logged_in? || valid_key?(params[:key])
   end
 
+  def new
+    if logged_in?
+      render_json errors: ["already logged in - log out before creating a new account"]
+      return
+    end
+    new_username = params[:new_username].downcase
+    user = User.new username: new_username, display_name: new_username,
+                     is_admin: false, status: User::ACTIVE_STATUS, email: params[:email],
+                     security_question: params[:security_question], security_answer: params[:security_answer]
+    if !user.valid?
+      render_json errors: ['user is not valid - probably missing required fields']
+      return
+    else
+      if User.where(username: new_username).exists?
+        user.errors.add :username, 'already exists'
+      end
+      if params[:new_password].length < 6
```

Comment:
might be worth factoring out this test into one place (maybe `user`?) since i expect we have this check in multiple places.